src/sentry_core.c

```diff
@@ -72,7 +73,15 @@ sentry__should_skip_upload(void)
 int
 sentry_init(sentry_options_t *options)
 {
-    sentry_close();
+    // this function is to be called only once, so we do not allow more than
+    // one call to be active. If the function is called twice, create an error
+    // and return fail (1)
+    sentry__mutex_lock(&g_init_lock);
+    if (g_options != NULL) {
+        SENTRY_ERROR("sentry_init() may only be called once");
+        sentry__mutex_unlock(&g_init_lock);
+        return 1;
```

Comment:
you leak `options` here. better to `goto fail` here as well.

Reply:
If the function is responsible to remove the given object, yes. When jumping to fail this is also tackled with this.